crates/weaver_resolver/src/registry.rs

```diff
@@ -115,9 +133,97 @@ pub fn resolve_semconv_registry(
         })
         .collect();
 
+    // Check the `any_of` constraints.
+    let attr_name_index = attr_catalog.attribute_name_index();
+    check_any_of_constraints(&ureg.registry, &attr_name_index)?;
+
+    // All constraints are satisfied.
+    // Remove the constraints from the resolved registry.
+    for group in ureg.registry.groups.iter_mut() {
+        group.constraints.clear();
+    }
+
     Ok(ureg.registry)
 }
 
+/// Checks the `any_of` constraints in the given registry.
+///
+/// # Arguments
+///
+/// * `registry` - The registry to check.
+/// * `attr_name_index` - The index of attribute names (catalog).
+///
+/// # Returns
+///
+/// This function returns `Ok(())` if all the `any_of` constraints are satisfied.
+/// Otherwise, it returns the error `Error::UnsatisfiedAnyOfConstraint`.
+pub fn check_any_of_constraints(
+    registry: &Registry,
+    attr_name_index: &[String],
+) -> Result<(), Error> {
+    for group in registry.groups.iter() {
+        // Build a list of attribute names for the group.
+        let mut group_attr_names = HashSet::new();
+        for attr_ref in group.attributes.iter() {
+            let attr_name =
+                attr_name_index
+                    .get(attr_ref.0 as usize)
+                    .ok_or(Error::UnresolvedAttribute {
+                        attribute_ref: *attr_ref,
+                    })?;
```

Comment:
`?` is sequential right now.  I don't think it's worth doing now, but we should look into attempting to find a way to detect errors in "parallel".  If we can, it's best to give users as many *valid* errors as we can at once.

Reply:
I agree, and I've already started to follow this approach in some other areas. However, it has not been implemented universally, particularly not here :-). It would probably be better to implement this sooner rather than later, so I will most likely work on it today.

Reply:
I improved the error handling to accumulate multiple errors.